### PR TITLE
[#937] Make all tests throw generic Exception

### LINKED ISSUE #937: Test: use generic throw Exception in all tests
The `repoConfig_overrideKeyword_success` test case has the `HelpScreenException` specified in its `throws` clause.

However, `HelpScreenException` is not thrown by the code in the body of the test case.
We should remove this exception from the `throws` clause.

image

### PR DESCRIPTION
**Pull request by FH-30**
Resolves #937.

Commit Message:
```
Make all tests throw generic Exception

The current test code for RepoSense contains throws clauses that do
not throw a generic Exception. This can result in a series of code
changes that have to be made when the exception(s) thrown by an
existing method used in the test code is changed, to prevent a
compiler error.

Let's make all methods in the test (and systemtest) folder(s) of
RepoSense throw a generic Exception.
```

### CODE REVIEW

**Comment by HCY123902** [comment]
LGTM. Just 1 minor suggestion. Although there are not many lines affected, you can actually split this into 2 separate pull requests because it seems that this pull request is handling 2 separate issues. The reason why pull request needs to be atmoic is that this can help to keep a clear record of what work is done for each individual issue and enable future contributors to know the previous progress

**Comment by FH-30** [comment]
Thanks @HCY123902 for the suggestion, I have undone the removal of redundant throws clauses in this PR and have made another PR for it [here](https://github.com/reposense/RepoSense/pull/1567).

File: src/test/java/reposense/parser/ArgsParserTest.java [hunk 1]
```diff
@@ -544,7 +544,7 @@ public void parse_repoAliases_sameResult() throws Exception {
 
     @Test (expected = ParseException.class)
     public void parse_noValidRepoLocation_throwsParseException()
-            throws ParseException, HelpScreenException {
+            throws Exception {
```
Comment (dcshzj) [inline src/test/java/reposense/parser/ArgsParserTest.java:547]: I think this can fit into one line.

**Review by dcshzj — CHANGES_REQUESTED** [review]

File: src/test/java/reposense/parser/ArgsParserTest.java (hunk 1 above)
Comment (FH-30) [reply src/test/java/reposense/parser/ArgsParserTest.java:547]: Right, thanks for pointing it out!

**Comment by FH-30** [comment]
Hi @HCY123902 and @dcshzj, any updates in this PR?

**Review by HCY123902 — APPROVED** [review]
LGTM

**Comment by HCY123902** [comment]
@dcshzj

**Review by dcshzj — APPROVED** [review]

**Review by fzdy1914 — APPROVED** [review]

**Pull request merged**